### ISSUE TITLE
Matrixexponential: "expm(0) == nan"

### ISSUE THREAD

**Issue by Jakob-Unfried** [google/jax#2131]
I have noticed that feeding an array of only zeros into `jax.scipy.linalg.expm` results in a purely `np.nan` array.
It should be the unit matrix, e.g. `expm(zeros) == np.eye(*zeros.shape)`

Minimal reproducer:
```python
import jax.numpy as np
from jax.version import __version__
from jax.scipy.linalg import expm

print(f'jax version: {__version__}')
#  jax version: 0.1.58

zero = np.zeros([2, 2], dtype=np.float32)
exp_zero = expm(zero)
print(f'exp(zero) == \n{exp_zero}')
#  exp(zero) == 
#  [[nan nan]
#  [nan nan]]

```

**Comment by Jakob-Unfried** [comment]
A dirty fix would be
```python
if np.allclose(A, np.zeros_like(A)):
     return np.eye(A.shape)
```

**Comment by terhorst** [comment]
The problem is with this line:

https://github.com/google/jax/blob/b2ef5bc09552e8ed39759df4ff49ea97e32db708/jax/scipy/linalg.py#L292

If `A_L1 == 0` then `np.floor_divide(np.log2(A_L1 / maxnorm), 1)` becomes `NaN`. A fix would be to rewrite it as 
```
n_squarings = np.maximum(0, np.floor(A_L1 / maxnorm)) 
```
Note that the same bug occurs a few lines below as well.

The division by one is weird, I wonder if the original author had some trick in mind?

**Comment by Jakob-Unfried** [comment]
@terhorst 

Good find!
Can confirm that this causes the problem.

Do you want to create a PR?
I would appreciate to see a fix somewhat soon, but i don't want to steal your thunder if you were going to do it anyway

**Comment by shoyer** [comment]
+ @jhueckelheim @sriharikrishna

**Comment by sriharikrishna** [comment]
Yes. I agree that this was a bug and that the suggested fix should be mage. Thanks to @Jakob-Unfried and @terhorst.

I believe that I might used `np.floor_divide(np.log2(A_L1 / maxnorm), 1)` because I was experiencing type promotion for n_squarings with my original logic. The use of floor() in the current code does not result in any type errors. I will push the fix shortly.

**Comment by shoyer** [comment]
Should be fixed now by #2192
